src/modules/75-cd/components/PipelineSteps/AzureWebAppServiceSpec/AzureWebAppStartupScriptSelection/__tests__/StartupScriptListView.test.tsx

```diff
@@ -47,15 +47,16 @@ describe('StartupScriptListView', () => {
         <StartupScriptListView {...propListView} />
       </TestWrapper>
     )
-    let gitConnector = await findByText(container, 'Github2')
+    const gitConnector = await findByText(container, 'Github2')
     expect(gitConnector).toBeTruthy()
     const deleteBtn = container.querySelector('[data-icon="main-trash"]') as Element
     expect(deleteBtn).toBeDefined()
     fireEvent.click(deleteBtn)
 
-    //should be closed
-    gitConnector = await findByText(container, 'Github2')
-    waitFor(() => expect(gitConnector).not.toBeTruthy())
+    setTimeout(() => {
+      //should be closed
+      waitFor(() => expect(gitConnector).not.toBeTruthy())
```

Comment:
`await` must always be used before `waitFor`

```suggestion
      await waitFor(() => expect(gitConnector).not.toBeTruthy())
```
---
Try removing the `setTimeout` after using `await` to check if it's still flaky.

Generally it's better to wait until something happens instead of a random timeout. For example, if the deletion happens after sending a request, you can spy on the function and wait until it's resolved. If there's no other way to fix this other than using a timeout, then you can pass timeout as an [option](https://testing-library.com/docs/dom-testing-library/api-async/#waitfor) to `waitFor` instead of using `setTimeout`.

Reply:
wow thanks, great info! will try with await first and lets see if the flakiness is solved